src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -154,9 +154,9 @@ private void consume() throws Exception {
         continue;
       }
       int partition = record.partition();
-      long index = (Long) avroRecord.get(DefaultTopicSchema.INDEX_FIELD.name());
+      long index = (long) avroRecord.get(DefaultTopicSchema.INDEX_FIELD.name());
```

Comment:
Have you tested that you can convert Ojbect to long? I am not sure if it works. See http://stackoverflow.com/questions/2351189/cannot-convert-from-type-object-to-long.